provider/core/src/baked/zerotrie.rs

```diff
@@ -107,6 +107,52 @@ impl<M: DataMarker> super::DataStore<M> for Data<M> {
     }
 }
 
+/// Regular baked data: a trie for lookups and a slice of values
+#[derive(Debug)]
+pub struct DataRef<M: DataMarker> {
+    // Unsafe invariant: actual values contained MUST be valid indices into `values`
```

Comment:
question: must this be a safety invariant?

It's somewhat upheld, but it being upheld assumes the integrity of ZeroTrie code; which is very nontrivial to check. If I faced this as an unsafe reviewer I would probably give up. ZeroTrie does not document its behavior as being a safety-usable guarantee.

It's a single unlikely branch, it should round off to nothing,

Reply:
this is copied from `pub struct Data<M: DataMarker>` above

Reply:
Oh. Hm. In that case I'm fine landing this, but we should discuss this unsafety. I'll file a followup.